### DISCUSSION TITLE
DOCUMENTATION: Add FAQ for each chapter

### DISCUSSION THREAD

**Discussion by LBoullosa** [hassanhabib/The-Standard#221] [General]
Hi!

Having in mind the Brokers FAQ could be helpful add as many FAQs for each chapter. This will be very useful when start learning The Standard, it could be a good starting point for anyone because I think, in some way, all of us, we always ask same similar questions about integrating components, what could be done or not when integrating components or any other concerns related to design a system according The Standard.

Firstly, I will enumerate a few questions I have, then I will post it here.

If anyone has questions. no matter if they are pretty simple, easy it will be welcomed for sure. Who knows, maybe one of these easy questions may help me or another. It will be great to add them to the documentation.

Thanks Hassan and everyone!

**Comment by hassanhabib** [comment]
> Hi!
> 
> Having in mind the Brokers FAQ could be helpful add as many FAQs for each chapter. This will be very useful when start learning The Standard, it could be a good starting point for anyone because I think, in some way, all of us, we always ask same similar questions about integrating components, what could be done or not when integrating components or any other concerns related to design a system according The Standard.
> 
> Firstly, I will enumerate a few questions I have, then I will post it here.
> 
> If anyone has questions. no matter if they are pretty simple, easy it will be welcomed for sure. Who knows, maybe one of these easy questions may help me or another. It will be great to add them to the documentation.
> 
> Thanks Hassan and everyone!

@LBoullosa fully agree.

**Comment by LBoullosa** [comment]
Hi! At this moment I collected this possible questions,

1) A Support Broker doesn´t require a Foundation Service. But,
Can we create a Foundation Service on top of a Support Broker?
Can we create a Processing Service on top of a Support Service?
In this scenario, a Support Foundation Service/Processing Foundation Service could be added anywhere like we add as a dependency to a Support Broker from a Processing, Orchestration or any other service type or even from an Exposure?

2) Designing a new library according the standard. For example, a library for dealing with translations somewhere (resource file, assembly, web api...), when integrating this library to a new project using The Standard,
should we create a broker to consume this library through his exposer or if there is no exposer could we use its Service directly?

3) When catching Exceptions inside XXXXXService.Exceptions, should we always call to CreateAndLogYYYYYYException within all the points inside the same flow execution when rethrowing/wrapping into a new Exception?

4) Multiple contracts, adapting/integrating between multiple contracts to fulfill the single contract rule.

**Comment by cjdutoit** [comment]
@LBoullosa / @hassanhabib we can add all questions to the Q&A section under discussions and then just add links in the FAQ section to the Q&A discussion.

**Reply by LBoullosa** [reply]
Sounds good to me!

**Reply by hassanhabib** [reply]
@cjdutoit sounds good. Good idea.
